src/components/expression/expression-renderer.js

```diff
@@ -487,7 +623,23 @@ genotet.ExpressionRenderer.prototype.dataLoaded = function() {
       i++;
     }
   }
+
+  i = 0;
+  profileCount = this.data.tfaProfiles.length;
+  while (i < profileCount) {
+    var geneIndex = heatmapData.geneNames.indexOf(
```

Comment:
This is poor implementation. You are querying the geneNames indices in two passes using this non-efficient indexOf. You can create a DS that stores the indices for heatmap.geneNames and perform the index retrieval much more efficiently.